packages/calypso-e2e/src/element-helper.ts

```diff
@@ -41,41 +47,57 @@ export async function waitForElementEnabled(
 /**
  * Locates and clicks on a specified tab on the NavTab.
  *
- * NavTabs are used throughout calypso to contain multiple related but separate pages within the same
- * overall page. For instance, on the Media gallery page a NavTab is used to filter the gallery to
+ * NavTabs are used throughout calypso to contain sub-pages within the parent page.
+ * For instance, on the Media gallery page a NavTab is used to filter the gallery to
  * show a specific category of gallery items.
  *
  * @param {Page} page Underlying page on which interactions take place.
  * @param {string} name Name of the tab to be clicked.
+ * @throws {Error} If the tab name is not the active tab.
  */
 export async function clickNavTab( page: Page, name: string ): Promise< void > {
-	// Mobile view - navtabs become a dropdown.
+	const selectedTabLocator = page.locator( selectors.navTabItem( { selected: true } ) );
+	const selectedTabName = await selectedTabLocator.innerText();
+
+	// Short circuit operation if the active tab and target tabs are the same.
+	// Strip numerals from the extracted tab name to account for the slightly
+	// different implementation in PostsPage.
+	if ( selectedTabName.replace( /[0-9]/g, '' ) === name ) {
```

Comment:
This is seen in the later section of this file as well.

Within `PostsPage`, the navtabs display the number of posts that match the category denoted by the navtab:
![image](https://user-images.githubusercontent.com/6549265/172502141-d0ab7b7e-7e1a-4437-82f7-b25347476a5d.png)

Since there would be no situation (as of writing) that calls for an arabic numeral in as part of the navtab name, I felt that this was an acceptable call to make.